aspen/request_processor/algorithm.py

```diff
@@ -70,6 +70,11 @@ def load_resource_from_filesystem(request_processor, dispatch_result):
     return {'resource': resources.get(request_processor, dispatch_result.match)}
 
 
-def render_resource(state, request_processor, resource):
-    state.setdefault('output', output.Output(charset=request_processor.charset_dynamic))
+def render_resource(state, resource):
     return {'output': resource.render(state)}
+
+
+def encode_output(output, request_processor):
+    if isinstance(output.body, unicode):
+        output.charset = request_processor.charset_dynamic
```

Comment:
> Is the word "charset" the problem?

No, `charset` is fine, it's what the `Content-Type` header uses.

> How about `encoding_{dynamic,static}`?

The purpose of renaming one or both of the settings is to distinguish them and make them easier to understand without having to read the docs. This proposal does not accomplish that.

As you pointed out at the beginning of this thread, `charset_dynamic` isn't limited to dynamic output, so to lift the confusion we want a name without `dynamic` in it.

Another solution is to push `charset_dynamic` up to pando, because its `use-aspen` branch ended up not using aspen's `encode_output` algorithm function (it does the encoding itself instead).